### PR TITLE
Add TCG Card Shop Simulator

### PR DESCRIPTION
**Pull request by SaxbyMod**
This may need some patching but this should do the trick.

### CODE REVIEW

File: src/assets/images/game_selection/TCG Card Shop Simulator.png
Comment (anttimaki) [inline src/assets/images/game_selection/TCG Card Shop Simulator.png:1]: Please used the attached image (it was approved by our UX department)
tcg-card-shop-simulator-cover-360x480

File: src/model/game/GameManager.ts [hunk 1]
```diff
@@ -794,6 +794,12 @@ export default class GameManager {
             "https://thunderstore.io/c/gorebox/api/v1/package-listing-index/", EXCLUSIONS,
             [new StorePlatformMetadata(StorePlatform.STEAM, "2027330")], "GoreBox.png",
             GameSelectionDisplayMode.VISIBLE, GameInstanceType.GAME, PackageLoader.BEPINEX, [""]),
+        
+        new Game("TCG Card Shop Simulator", "TCG Card Shop Simulator", "TCG Card Shop Simulator",
```
Comment (anttimaki) [inline src/model/game/GameManager.ts:798]: The `internalFolderName` parameter does not match the value used in `InstallationRuleApplicator.ts`, preventing the app from building.

File: src/model/game/GameManager.ts [hunk 2]
```diff
@@ -794,6 +794,12 @@ export default class GameManager {
...
+            "TCG Card Shop Simulator", ["Card Shop Simulator.exe"], "Card Shop Simulator_Data",
+            "thunderstore.io/c/tcg-card-shop-simulator/api/v1/package/", "https://raw.githubusercontent.com/ebkr/r2modmanPlus/master/modExclusions.md",
```
Comment (anttimaki) [inline src/model/game/GameManager.ts:800]: - The `tsUrl` parameter is outdated (see above for an example)
- Use the `EXCLUSIONS` const for `exclusionUrl` parameter

File: src/model/game/GameManager.ts [hunk 3]
```diff
@@ -794,6 +794,12 @@ export default class GameManager {
...
+            [new StorePlatformMetadata(StorePlatform.STEAM, "3070070"), new StorePlatformMetadata(StorePlatform.OTHER)], "TCG Card Shop Simulator.png",
```
Comment (anttimaki) [inline src/model/game/GameManager.ts:801]: - What is the other platform the game is available on?
- The convention is for the `gameImage` parameter to not include spaces

File: src/r2mm/installing/default_installation_rules/InstallationRuleApplicator.ts [hunk 4]
```diff
@@ -149,6 +149,7 @@ export default class InstallationRuleApplicator {
             buildBepInExRules("Distance"),
             buildBepInExRules("FiveNightsAtFreddysIntoThePit"),
             buildBepInExRules("GoreBox"),
+            buildBepInExRules("TCG Card Shop Simulator"),
```
Comment (anttimaki) [inline src/r2mm/installing/default_installation_rules/InstallationRuleApplicator.ts:152]: The string value used here doesn't match the one used in `InstallationRuleApplicator.ts`, preventing the app from building.

**Review by anttimaki — CHANGES_REQUESTED** [review]

File: src/assets/data/games.json [hunk 5]
```diff
@@ -5452,6 +5452,80 @@
       },
       "discordUrl": "https://discord.gg/sunhaven"
     },
+    "tcg-card-shop-simulator": {
```
[the remark this answers is not in the document]
Comment (SaxbyMod) [reply src/assets/data/games.json:5455]: Gotcha didnt know ^^

File: src/model/game/GameManager.ts (hunk 2 above)
Comment (SaxbyMod) [reply src/model/game/GameManager.ts:800]: Will fix

File: src/model/game/GameManager.ts (hunk 3 above)
Comment (SaxbyMod) [reply src/model/game/GameManager.ts:801]: First one was likely an oversight on my end Inhad that there in the case that other platforms would pop up similar to Inscryption. Also understood will change the image name.

File: src/r2mm/installing/default_installation_rules/InstallationRuleApplicator.ts (hunk 4 above)
Comment (SaxbyMod) [reply src/r2mm/installing/default_installation_rules/InstallationRuleApplicator.ts]: Should be resolved now

File: src/model/game/GameManager.ts (hunk 3 above)
Comment (SaxbyMod) [reply src/model/game/GameManager.ts]: Should be resolved now [Keeping the Other]

File: src/model/game/GameManager.ts (hunk 1 above)
Comment (SaxbyMod) [reply src/model/game/GameManager.ts]: Should be resolved now

File: src/assets/images/game_selection/TCG Card Shop Simulator.png
Comment (SaxbyMod) [reply src/assets/images/game_selection/TCG Card Shop Simulator.png:1]: Should be resolved now

File: src/assets/data/games.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (SaxbyMod) [reply src/assets/data/games.json]: Should be resolved now

File: src/model/game/GameManager.ts [hunk 6]
```diff
@@ -794,6 +794,12 @@ export default class GameManager {
             "https://thunderstore.io/c/gorebox/api/v1/package-listing-index/", EXCLUSIONS,
             [new StorePlatformMetadata(StorePlatform.STEAM, "2027330")], "GoreBox.png",
             GameSelectionDisplayMode.VISIBLE, GameInstanceType.GAME, PackageLoader.BEPINEX, [""]),
+        
+        new Game("TCG Card Shop Simulator", "TCGCardShopSimulator", "TCG Card Shop Simulator",
+            "TCG Card Shop Simulator", ["Card Shop Simulator.exe"], "Card Shop Simulator_Data",
+            "https://thunderstore.io/c/tcg-card-shop-simulator/api/v1/package-listing-index/", EXCLUSIONS,
+            [new StorePlatformMetadata(StorePlatform.STEAM, "3070070"), new StorePlatformMetadata(StorePlatform.OTHER)], "TCGCardShopSimulator.png",
```
Comment (anttimaki) [inline src/model/game/GameManager.ts:801]: What is the other platform? How do I test it?

**Review by anttimaki — COMMENTED** [review]
One open question, otherwise the PR looks good now.

I also left a change request on the ecosystem schema PR, which needs to be fixed, and the PR merged, for me to be able to test this properly.

File: src/model/game/GameManager.ts (hunk 6 above)
Comment (SaxbyMod) [reply src/model/game/GameManager.ts:801]: Other was left behind as a case of there being a potential other platform in the future but if prefered I can remove it ^^

Comment (SaxbyMod) [reply src/model/game/GameManager.ts]: Fixed

**Comment by SaxbyMod** [comment]
> One open question, otherwise the PR looks good now.
> 
> I also left a change request on the ecosystem schema PR, which needs to be fixed, and the PR merged, for me to be able to test this properly.

All Issues should now be fixed!

**Review by anttimaki — APPROVED** [review]

**Pull request merged**